examples/ansible/setup_libvirt_host.yml

```diff
@@ -1,129 +1,120 @@
-        state: present
-      register: required_pkgs_install
-      when:
-        - ((workload.required_pkgs | default([])) | length) > 0
-
-    - name: Handle reboot, if needed
-      when:
-        - required_pkgs_install is changed
-      block:
-
-        - name: Print message requesting immediate reboot
-          ansible.builtin.debug:
-            msg: >-
-              Required system packages have been installed via
-              transactional-update to support workload
-              {{ workload.name }}.
-              An immediate reboot is required to activate these
-              packages before proceding. Once reboot has completed
-              please re-run this playbook.
-
-        - name: Request immediate reboot  # noqa no-handler
-          ansible.builtin.fail:
-            msg: "Reboot required"
-
-    - name: Setup service for workload {{ workload.name }}
-      when:
-        - workload.service is defined
-      block:
-
-        - name: Ensure service can be started for workload {{ workload.name }}
-          become: true
-          ansible.builtin.systemd_service:
-            name: "{{ workload.service.name }}"
-            state: "started"
-          register: workload_service_start
-
-        - name: Enable service to start on reboot for workload {{ workload.name }}
-          become: true
-          ansible.builtin.systemd_service:
-            name: "{{ workload.service.name }}"
-            enabled: true
-          when:
-            - workload_service_start is success
-
-        - name: Wait for service to be ready for workload {{ workload.name }}
-          when:
-            - workload.service.wait_for is defined
-          block:
-
-            - name: Check service path for workload {{ workload.name }}
-              ansible.builtin.wait_for:
-                path: "{{ workload.service.wait_for.path }}"
-              when:
-                - workload.service.wait_for.path is defined
-
-            - name: Check service port for workload {{ workload.name }}
-              ansible.builtin.wait_for:
-                port: "{{ workload.service.wait_for.port }}"
-              when:
-                - workload.service.wait_for.port is defined
-
-#
-# Verify basic operation of the Libvirt service
-#
-
-- name: Verify that Libvirt is operational
-  hosts: alphost
-  tasks:
-
-    - name: Query running Libvirt VMs
-      become: true
-      community.libvirt.virt:
-        command: list_vms
-      register: virt_list_vms
+    - name: Install tools for kvm-client
+      ansible.builtin.command: >-
+        podman container runlabel install registry.opensuse.org/suse/alp/workloads/tumbleweed_containerfiles/suse/alp/workloads/kvm-client:latest
+      register: workload_runlabel_install
+      changed_when:
+        - ('already exist' not in workload_runlabel_install.stdout)
 
-    - name: Report running Libvirt VMs, if any
+    - name: Ensure libvirtd is stopped and disabled
+      ansible.builtin.service:
```

Comment:
Note that you have switched to the generic service action which should detect and use the underlying systemd_service action to do the work; given that we know that target system is systemd based, I had opted to be explicit and use the systemd_service action. ;-)